### ISSUE TITLE
always linked against OpenCV 3.2 *and* 3.3 (not found)

### ISSUE THREAD

**Issue by HWiese1980** [FORTH-ModelBasedTracker/PyOpenPose#9]
Hi,

I'm successfully building PyOpenPose on my system but cannot use it because for some reasons I don't see it always gets linked against OpenCV 3.2 *and* OpenCV 3.3, the latter which does not exist on my machine. I've already tried to set 3.2 explicitly in `CMakeLists.txt` but the links to 3.3 are still there. Iimporting the module in Python fails because it can't find the libraries. 

Why does it insist on linking against OpenCV 3.3 even if I set it explicitly to 3.2? 

Thanks

**Comment by padeler** [comment]
If 3.3 is not on your machine it cannot be linked with it (the linker would fail) so the problem is somewhere else. In any case PyOpenPose does not have any explicit requirements for opencv3.3.

**Comment by HWiese1980** [comment]
This is truly weird. I can successfully build it. But upon `import` it complains about missing libraries `libopencv_core.so.3.3` and others. `ldd` also states that they are needed but not found. 

Creating symlinks for the missing libraries pointing to the corresponding `libopencv_....so` helps as a workaround but cannot be the royal road. 

If you call `ldd` on your `PyOpenPose.so`, does it also show the OpenCV 3.3 dependencies? 

Thanks

**Comment by padeler** [comment]
No it does not because i do not have opencv3.3 on my system (i have opencv3.2). Try deleting your cmake cache and recreate it. It looks like cmake has kept some cached info about opencv3.3 that are no longer valid along side with the opencv3.2 which is actually used for linking.

**Comment by toinsson** [comment]
Hello, 
@HWiese1980, did you find the origin of the problem? I am stuck on the same issue. Can do the symlinking but indeed it feels hacky.

**Comment by HWiese1980** [comment]
Well, I manually removed all the **related** libraries from /usr/local/lib, recompiled everything from scratch (including OpenCV and all other dependencies) and it went away. I assume that the reason was some dependency artifact somewhere along the branches of the dependency tree. I haven't found the actual cause though...

**Comment by toinsson** [comment]
Found the problem: `openpose` did not link to dependent library in the first place. This left undefined link in the lib when checking with `ldd`:

	libopencv_objdetect.so.3.1 => not found
	libopencv_highgui.so.3.1 => not found
	libopencv_videoio.so.3.1 => not found
	libopencv_imgcodecs.so.3.1 => not found
	libopencv_imgproc.so.3.1 => not found
	libopencv_core.so.3.1 => not found

Eventually, when linking PyOpenPose, some would be updated other not.
I added CMAKE_INSTALL_RPATH_USE_LINK_PATH to the compile flags and it solved the linking issue in `openpose`, and thus in `PyOpenPose` too.
